text/2141-alternative-registries.md

```diff
@@ -43,18 +43,18 @@ In order to tell Cargo about a registry other than crates.io, you can specify an
 
 ```toml
 [registries]
-choose-a-name = "https://my-intranet:8080/index"
+my-registry = "https://my-intranet:8080/index"
 ```
 
-Instead of `choose-a-name`, place the name you'd like to use to refer to this registry in your
-`Cargo.toml` files. The URL specified should contain the location of the registry index for this
-registry; the registry format is specified in the [Registry Index Format Specification
-section][registry-index-format-specification].
+Every registry has a name, specified in the `config.json` file in its index. The name specified
+must match the registry's configured name. The URL specified should contain the location of the
```

Comment:
Requiring every registry to have a unique name will cause problems as outlined by alex in [this comment](https://github.com/rust-lang/cargo/issues/4880#issuecomment-358483209):

> Now one thing I don't like about custom names though is that it's possible to have two disjoint communities that both name their registry foo. This in turn can cause confusion and is a symptom of the fact that there's no one arbitrarting handing out names of registries.